### PR TITLE
get coprecessing modes

### PR DESCRIPTION
**Pull request by md-arif-shaikh**

### CODE REVIEW

**Comment by md-arif-shaikh** [comment]
@vijayvarma392 this is now ready to review. It's fairly simple PR. I just added functions in `load_data.py` to get modes in `coprecessing` frame for input `dataDict` in the `inertial` frame. In addition, I changed the Python setup in the GitHub workflow so that `scri` is installed without any problem.

File: gw_eccentricity/load_data.py [hunk 1]
```diff
@@ -1383,6 +1385,108 @@ def get_num_orbits_duration_from_horizon_data(horizon_filepath, num_orbits):
...
+        data=ordered_mode_list,
+        ell_min=ell_min,
+        ell_max=ell_max,
+        frameType=scri.Inertial,
+        r_is_scaled_out=True,
+        m_is_scaled_out=True)
+
+    # co-precessing frame modes
+    w_coprecessing = deepcopy(w).to_coprecessing_frame()
+    return {"t": data_dict["t"], "hlm": unpack_scri_modes(deepcopy(w_coprecessing))}
```
Comment (vijayvarma392) [inline gw_eccentricity/load_data.py:1487]: Do we need to be worried about:
Sometimes the user may have other stuff in data_dict that they may have added for whatever reason. If you just replace the dict with what is returned here, data_dict would no longer have those.

Instead, what if we just update the "hlm" key of the input dict and return the dict?

**Review by vijayvarma392 — CHANGES_REQUESTED** [review]

File: gw_eccentricity/load_data.py [hunk 2]
```diff
@@ -1383,6 +1385,108 @@ def get_num_orbits_duration_from_horizon_data(horizon_filepath, num_orbits):
...
+    `ell`=2, the list should be [(2, -2), (2, 1), (2, 0), (2, 1), (2, 2)]
+    """
+    keys = modes_dict.keys()
+    shape = modes_dict[(2, 2)].shape
+    n_elem = (ell_max + 3) * (ell_max - 1)
+    result = np.zeros((shape[0], n_elem), dtype=np.complex128)
+    i = 0
+    for ell in range(ell_min, ell_max + 1):
+        for m in range(-ell, ell + 1):            
+            if (ell, m) in keys:
```
[the remark this answers is not in the document]
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:1417]: comment and checks added

File: gw_eccentricity/load_data.py (hunk 1 above)
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py]: Done

File: gw_eccentricity/load_data.py [hunk 3]
```diff
@@ -1484,7 +1493,11 @@ def get_coprecessing_data_dict(data_dict, ell_min=2, ell_max=2):
 
     # co-precessing frame modes
     w_coprecessing = deepcopy(w).to_coprecessing_frame()
-    return {"t": data_dict["t"], "hlm": unpack_scri_modes(deepcopy(w_coprecessing))}
+    # Create a copy of data_dict and replace the "hlm" modes in the inertial frame
+    # with the corresponding modes in the coprecessing frame
+    data_dict_copr = deepcopy(data_dict)
+    data_dict_copr.update({"hlm": unpack_scri_modes(deepcopy(w_coprecessing))})
+    return data_dict
```
[the remark this answers is not in the document]
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py]: Yes, corrected it. Yes, I have a notebook but checked only the other changes. I should have run the full notebook.

**Review by vijayvarma392 — APPROVED** [review]

**Pull request merged**